src/mono/browser/runtime/corebindings.c

```diff
@@ -84,15 +90,19 @@ void bindings_initialize_internals (void)
 	mono_add_internal_call ("Interop/Runtime::InvokeJSFunctionSend", mono_wasm_invoke_js_function_send);
 	mono_add_internal_call ("Interop/Runtime::CancelPromise", mono_wasm_cancel_promise);
 	mono_add_internal_call ("Interop/Runtime::CancelPromisePost", mono_wasm_cancel_promise_post);
-	mono_add_internal_call ("Interop/Runtime::SetEntryAssembly", mono_wasm_set_entry_assembly);
+	mono_add_internal_call ("Interop/Runtime::AssemblyGetEntryPoint", mono_wasm_assembly_get_entry_point);
```

Comment:
Do we need these icall definitions separate for threads and non threads?

Reply:
I thought this was more readable way, I can change it later

Reply:
There is a section that applies for both variants